src/Microsoft.AspNetCore.ResponseCompression/ResponseCompressionMiddleware.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO.Compression;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+using Microsoft.Net.Http.Headers;
+
+namespace Microsoft.AspNetCore.ResponseCompression
+{
+    public class ResponseCompressionMiddleware
+    {
+        private readonly RequestDelegate _next;
+
+        private readonly IResponseCompressionProvider[] _compressionProviders;
+
+        private readonly HashSet<string> _mimeTypes;
+
+        /// <summary>
+        /// Initialize the Response Compression middleware.
+        /// </summary>
+        /// <param name="next"></param>
+        /// <param name="options"></param>
+        public ResponseCompressionMiddleware(RequestDelegate next, IOptions<ResponseCompressionOptions> options)
+        {
+            if (options.Value.MimeTypes == null)
+            {
+                throw new ArgumentNullException(nameof(options.Value.MimeTypes));
```

Comment:
I'm not sure about this exception. `MimeTypes` is not one of the constructor's parameters. This will cause confusion. Instead how about throw `ArgumentException($"{nameof(options.Value.MimeTypes)} is not provided in argument {nameof(options)}")`